imports/ui/layouts/AppContent.jsx

```diff
@@ -0,0 +1,117 @@
+import React, { useEffect } from 'react';
+import { Switch, Route } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import i18n from 'meteor/universe:i18n';
+import { TransitionGroup, CSSTransition } from 'react-transition-group';
+import UserMenu from '../components/UserMenu.jsx';
+import ListList from '../components/ListList.jsx';
+import LanguageToggle from '../components/LanguageToggle.jsx';
+import ConnectionNotification from '../components/ConnectionNotification.jsx';
+import Loading from '../components/Loading.jsx';
+import ListPageContainer from '../containers/ListPageContainer.jsx';
+import AuthPageSignIn from '../pages/AuthPageSignIn.jsx';
+import AuthPageJoin from '../pages/AuthPageJoin.jsx';
+import NotFoundPage from '../pages/NotFoundPage.jsx';
+import { useMenuOpen } from '../state/MenuOpenState.jsx';
+import { useLocale } from '../state/LocaleState.jsx';
+
+const AppContent = ({
+  connexionNotification,
+  lists,
+  loading,
+  location,
+  logout,
+  user,
+}) => {
+  const [menuOpen, setMenuOpen] = useMenuOpen();
+  const [, setLocale] = useLocale();
+
+  const handleLocaleChange = (newLocale) => {
+    setLocale(newLocale);
+  };
+
+  useEffect(() => {
+    i18n.onChangeLocale(handleLocaleChange);
+    return () => {
+      i18n.offChangeLocale(handleLocaleChange);
+    };
+  }, []);
+
+  const closeMenu = () => {
+    setMenuOpen(false);
+  };
+  const transitionMenuOpen = {
+    get: () => menuOpen,
+    set: newMenuOpen => setMenuOpen(newMenuOpen),
+  };
+  const commonChildProps = {
+    menuOpen: transitionMenuOpen,
+  };
+
+  return (
+    <div id="container" className={menuOpen ? 'menu-open' : ''}>
+      <section id="menu">
+        <LanguageToggle />
+        <UserMenu user={user} logout={logout} />
+        <ListList lists={lists} />
+      </section>
+      {connexionNotification
+        ? <ConnectionNotification />
+        : null}
+      <div className="content-overlay" onClick={closeMenu} />
+      <div id="content-container">
+        {loading ? (
+          <Loading key="loading" />
+        ) : (
+          <TransitionGroup>
+            <CSSTransition
+              key={location.key}
+              classNames="fade"
+              timeout={200}
+            >
+              <Switch location={location}>
+                <Route
+                  path="/lists/:id"
+                  render={({ match }) => (
+                    <ListPageContainer match={match} {...commonChildProps} />
+                  )}
+                />
+                <Route
+                  path="/signin"
+                  render={() => <AuthPageSignIn {...commonChildProps} />}
+                />
+                <Route
+                  path="/join"
+                  render={() => <AuthPageJoin {...commonChildProps} />}
+                />
+                <Route
+                  path="/*"
+                  render={() => <NotFoundPage {...commonChildProps} />}
+                />
```

Comment:
If we manage to remove the `commonChildProps`, this can be changed to use the `component` prop instead of `render` like this:

```jsx
<Route path="/join" component={AuthPageJoin} />
```